fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBDatabase.java

```diff
@@ -784,7 +793,9 @@ public FDBDatabaseRunner newRunner() {
      */
     @Nonnull
     public FDBDatabaseRunner newRunner(@Nullable FDBStoreTimer timer, @Nullable Map<String, String> mdcContext) {
-        return new FDBDatabaseRunnerImpl(this, timer, mdcContext);
+        FDBDatabaseRunner runner = newRunner();
+        runner.getContextConfigBuilder().setTimer(timer).setMdcContext(mdcContext);
+        return runner;
```

Comment:
```suggestion
        return newRunner(timer, mdcContext, null);
```
I think. I can make arguments that the other two `newRunner`s should remain as is, but this one seems like it should populate the default value of the other one.

Reply:
How so? The default for `weakSemantics` is `null`, right?

Reply:
Right, sorry. I meant just that this method should call `newRunner(FDBStoreTimer, Map, WeakReadSemantics)` with a `null` `WeakReadSemantics` rather than tweaking the context config builder just so that it fits into the Java pattern for "methods with default parameters". But I don't feel strongly about this and could see this staying as is

Reply:
Oh, I see. I would just as soon deprecate those two methods. But either way, I don't think the one with all the arguments is, in fact, the most general one, like in that pattern. The one with one argument is, so maybe they should be expressed in terms of that?